File: web/src/models/AuthModels.ts
export interface JwtPayload {
  sub: string;
  email: string;
  exp: number;
  iss?: string;
  /** Dev-bypass mock token: a single CES role name. */
  role?: string;
  /** Keycloak token: raw client role names, mapped through KEYCLOAK_ROLE_MAP. */
  roles?: string[];
  name?: string;
  preferred_username?: string;
}

export interface User {
  id: string;
  email: string;
  roles: string[];
  displayName?: string;
}

/**
 * Response from `POST /api/auth/callback`. Carries no refresh token by design — that
 * stays in the encrypted HttpOnly cookie the API issues and is unreachable from JS.
 */
export interface AuthCallbackResponse {
  accessToken: string;
  /** Access-token lifetime in seconds. */
  expiresIn: number;
  /** Validated server-side; safe to navigate to. */
  returnUrl: string;
}

/** Response from `POST /api/auth/refresh`. Same no-refresh-token rule as the callback. */
export interface AuthRefreshResponse {
  accessToken: string;
  expiresIn: number;
}

/** Response from `POST /api/auth/logout`. */
export interface AuthLogoutResponse {
  endSessionUrl: string;
}

File: api/CES.API/Models/AuthModels.cs
using CES.API.Authentication;

namespace CES.API.Models
{
    /// <summary>What <c>AuthCallback.vue</c> posts after Keycloak redirects the browser back.</summary>
    public class AuthCallbackRequest
    {
        public string Code { get; set; } = string.Empty;
        public string State { get; set; } = string.Empty;
    }

    /// <summary>
    /// The callback response. Deliberately carries no refresh token — that stays in the
    /// encrypted HttpOnly cookie and must never reach JavaScript.
    /// </summary>
    public class AuthCallbackResponse
    {
        public string AccessToken { get; set; } = string.Empty;

        /// <summary>Access-token lifetime in seconds; the SPA schedules its renewal from this.</summary>
        public int ExpiresIn { get; set; }

        /// <summary>Validated at /api/auth/login and carried in the encrypted login cookie.</summary>
        public string ReturnUrl { get; set; } = AuthConstants.DefaultReturnUrl;
    }
}
